### DISCUSSION TITLE
How to use useWindowSize from VueUse with nuxt properly

### DISCUSSION THREAD

**Discussion by jd-solanki** [nuxt/nuxt#18206] [Questions]
Hi,

I am trying to use [`useWindowSize`](https://vueuse.org/core/useWindowSize/) composable from VueUse. The problem I am facing is I am unable to dynamically apply class-based on window width.

I already asked it in VueUse [community](https://github.com/vueuse/vueuse/issues/881) but got no luck.

How can I properly use this kind of composable to calculate window width and apply the dynamic class? 🤔 

**Demo** ✨ 

CodeSandbox: https://codesandbox.io/s/nuxt-vueuse-windowsize-query-328mo

👆🏻 In the above demo I want to apply `disabled` class if the window width is less than `1200` but it isn't working, even if I resize the window.

**Comment by wobsoriano** [comment]
Your code is correct but it looks like the server value of `width` is `Infinity`. So I guess you can check if it's on browser first?

```html
<script setup lang="ts">
const { width } = useWindowSize();
const isBrowser = process.client
</script>

<template>
  <div
    style="
      display: flex;
      align-items: center;
      justify-content: center;
      min-height: 100vh;
    "
  >
    <div v-if="isBrowser">
      <div class="box" :class="{ disabled: width < 1200 }" />
    </div>
  </div>
</template>
```

You'll get hydration warnings tho

EDIT: Setting `initialWidth` works perfectly tho.

```ts
const { width } = useWindowSize({ initialWidth: 0 });
```

**Reply by jd-solanki** [reply]
Thanks for the response. 

Setting `initialWidth` doesn't reflect when `window` object is found. So, in this case, setting `initialWidth` always stays `0`.

You can check it by setting:

```diff
- const { width } = useWindowSize();
+ const { width } = useWindowSize({ initialWidth: 0 });
```

```diff
- <div class="box" :class="{ disabled: width < 1200 }" />
+ <div class="box" :class="{ disabled: width > 1200 }" />
```

With this change even if we open the app in `> 1200` window `disabled` class doesn't apply.

Regards.
